File: 0804-unique-morse-code-words/0804-unique-morse-code-words.py
class Solution:
    def uniqueMorseRepresentations(self, words: List[str]) -> int:
        alpha = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]
        output = []
        for word in words:
            output.append("".join([alpha[ord(x)-97] for x in word]))
        return (len(list(set(output))))

File: 0804-unique-morse-code-words/0804-unique-morse-code-words.java
class Solution {
    public int uniqueMorseRepresentations(String[] words) {
        String[] alpha = {".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
		HashSet<String> unique = new HashSet();
		for(String ele : words){
		    StringBuilder morse = new StringBuilder();
		    for(int i=0; i<ele.length(); i++)
		        morse.append(alpha[(int)ele.charAt(i)-97]);
		    unique.add(morse.toString());
		}
		return unique.size();  
    }
}